src/ducks/recurrence/service.js

```diff
@@ -19,20 +19,31 @@ export const NB_DAYS_LOOKBACK = 100
 
 /**
  * Fetch transactions from 3 months back if there is already recurrences,
- * otherwise all transactions
+ * otherwise all transactions.
+ * We filter out transactions already associated with a recurrence as we don't
+ * want to change it.
+ *
  * @param {array} recurrences - Hydrated bundle of recurrences
  * @returns transactions
  */
 const makeQueryForTransactions = recurrences => {
   if (recurrences.length === 0) {
-    return Q(TRANSACTION_DOCTYPE)
+    // We still filter out transactions that were explicitely marked as not
+    // recurrent.
+    return Q(TRANSACTION_DOCTYPE).partialFilter({
+      'relationships.recurrence': { $exists: false }
+    })
```

Comment:
For this particular case, I'm not sure the partialFilter is actually worth it: this will create an index for probably few docs (the ones marked as not recurrent), and will use the `_find` endpoint instead of the `_all_docs` endpoint. And the later is actually more efficient

Reply:
As I don't have any information on the number of marked documents I guess it's a good bet to remove the partial index but we might have more marked documents than we think when there aren't any recurrence. Especially if the user deleted all their recurrences (all associated transactions are then marked as not recurrent).

Reply:
Ah, one more thing : removing the partial filter here means we'll need to loop over the fetched transactions to skip the few ones marked as not recurrent :/

Reply:
Well, if we can't estimate the potential number of marked documents, and therefore the actual impact, I would say it's your call :)
I tend to think the average case would be in favor of removing the partial filter, but I don't have any data to support this, and the need to loop over all the transactions afterwards to filter out the recurrences is cumbersome. So either way, it's fine by me